### PR TITLE
User login

### LINKED ISSUE #11: login API
Authentication API  by payer Id and password

### PR DESCRIPTION
**Pull request by Ebugo**
## Description
Worked on the user login endpoint. Returned access token on successful authentication. 
Dependency required is express-validator for user input validation.

All pull requests should be related to open issues. Indicate the issue(s) bellow and remove this line:
Fixes #11 #10 

## Checklist:

- [x ] My code follows the style guidelines of this project
- [x ] I have performed a self-review of my own code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [x ] I have added necessary inline code documentation
- [x ] I have added tests that prove my fix is effective and that this feature works
- [x ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by triple0t — DISMISSED** [review]

File: src/controllers/usersController.js [hunk 1]
```diff
@@ -19,10 +22,48 @@ exports.registeration = (req, res) => {
     res.json(err);
   });
 };
-exports.login = (req, res) => {
-// return token
-  res.json('login in progress');
-};
+
+exports.login = [
```
Comment (nextwebb) [inline src/controllers/usersController.js:27]: the issue was for users to login with their tax payer Id and  password . not email and password

File: src/controllers/usersController.js [hunk 2]
```diff
@@ -19,10 +22,48 @@ exports.registeration = (req, res) => {
...
+    check('email').isEmail().withMessage('Invalid email addres'),
+    // password must be at least 5 chars long
+    check('password').isLength({ min: 5 }).withMessage('Password must be at least 5 chars long')
+  ], async (req, res) => {
+    // Finds the validation errors in this request and wraps them in an object with handy functions
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(422).json({ status: false, errors: errors.array() });
+    }
+    const user = await User.findOne({
```
Comment (nextwebb) [inline src/controllers/usersController.js]: i cant see the findOne method as a change in the users class

File: src/controllers/usersController.js (hunk 1 above)
Comment (Ebugo) [reply src/controllers/usersController.js:27]: Ok, would fix that

File: src/controllers/usersController.js (hunk 2 above)
Comment (Ebugo) [reply src/controllers/usersController.js]: I don't get this...

File: src/controllers/usersController.js (hunk 1 above)
Comment (nextwebb) [reply src/controllers/usersController.js:27]: Okay

**Review by nextwebb — DISMISSED** [review]
LGTM 🚀

File: src/controllers/usersController.js (hunk 1 above)
Comment (Ebugo) [reply src/controllers/usersController.js:27]: fixed

**Review by nextwebb — APPROVED** [review]
okay

**Review by triple0t — APPROVED** [review]

**Pull request merged**